Describe how to use AWS and GCP resources (#56)
capactio/website: docs/example/typeinstances.md

> # Setup Type Instances with credentials
> 
> To use cloud resources like AWS RDS or Google Cloud SQL you need to configure Type Instances which will hold required credentials.

Reviewer: ```suggestion
# Common TypeInstances

This document lists all TypeInstances, which are commonly used across our examples. One of common use case is usage of cloud provider services. For example, to use managed database such as AWS RDS or Google Cloud SQL, you need to create TypeInstances which stores required credentials.

## Prerequisites
* [Capact CLI](../cli/getting-started.mdx) installed.
* Cluster with Capact installation. See the [installation tutorial](../installation/local.md). 
- [`jq`](https://stedolan.github.io/jq/download/) installed.
```

> ### AWS

Reviewer: ```suggestion
## AWS Credentials
```

1. Create a new User in AWS dashboard [https://console.aws.amazon.com/iam/home?region=eu-west-1#/home](https://console.aws.amazon.com/iam/home?region=eu-west-1#/home)

   - Add the **AdministratorAccess** permissions.
   - Note the access key and secret key.

1. Create a file with the AWS Credentials:

    ```bash
    cat > /tmp/aws-ti.yaml << ENDOFFILE
    typeInstances:
      - alias: aws-sa
        attributes:
          - path: cap.attribute.cloud.provider.aws
            revision: 0.1.0
        typeRef:
          path: cap.type.aws.auth.credentials
          revision: 0.1.0
        value:
          accessKeyID: {ACCESS_KEY}
          secretAccessKey: {SECRET_KEY}
    ENDOFFILE
    ```

> 1. Create AWS Credentials TypeInstance:
> 
>     ```bash
>     export TI_ID=$(capact typeinstance create -f /tmp/aws-ti.yaml -ojson | jq -r '.[].id')
>     ```

Reviewer: 🤔   I think it will be hard to avoid env variables conflicts in docs using the TypeInstance creation. I would recommend reworking this part to:

``````suggestion
1. Create AWS Credentials TypeInstance:

    ```bash
    capact typeinstance create -f /tmp/aws-ti.yaml -ojson
    ```

1. Note the ID.
``````

> ### GCP

Reviewer: ```suggestion
### GCP Service Account
```

1. Create a GCP Service Account JSON access key:
   

>    1. Open [https://console.cloud.google.com](https://console.cloud.google.com) and select your project.

Reviewer: ```suggestion
   1. Open [GCP Console](https://console.cloud.google.com) and select your project.
```

   
   2. In the left pane, go to **IAM & Admin** and select **Service accounts**.
   
   3. Click **Create service account**, name your account, and click **Create**.
   
   4. Click **Create key** and choose `JSON` as the key type.
   

>    5. Save the `JSON` file.

Reviewer: ```suggestion
   5. Save the `JSON` file under the `/tmp/gcp-sa.json` path.
```

   
   6. Click **Done**.

1. Create a file with the GCP Credentials:

>     ```yaml
>     cat > /tmp/gcp-sa-ti.yaml << ENDOFFILE
>     typeInstances:
>       - alias: gcp-sa
>         typeRef:
>           path: cap.type.gcp.auth.service-account
>           revision: 0.1.0
>         attributes:
>           - path: cap.attribute.cloud.provider.gcp
>             revision: 0.1.0
>         value: { # Put here your GCP Service Account JSON.
>           "type": "service_account",
>           [...]
>         }
>     ENDOFFILE
>     ```

Reviewer: What do you think about such commands:

``````suggestion
    ```bash
    export GCP_SA_VALUE=$(cat /tmp/gcp-sa.json | jq -c)
    cat > /tmp/gcp-sa-ti.yaml << ENDOFFILE
    typeInstances:
      - alias: gcp-sa
        typeRef:
          path: cap.type.gcp.auth.service-account
          revision: 0.1.0
        attributes:
          - path: cap.attribute.cloud.provider.gcp
            revision: 0.1.0
        value: ${GCP_SA_VALUE}
    ENDOFFILE
    ```
``````

> 1. Create a GCP Credentials TypeInstance:
>     ```bash
>     export TI_ID=$(capact typeinstance create -f /tmp/gcp-sa-ti.yaml -ojson | jq -r '.[].id')
>     ```
> 

Reviewer: Similarly as above:

``````suggestion
1. Create GCP Service Account TypeInstance:

    ```bash
    capact typeinstance create -f /tmp/gcp-sa-ti.yaml -ojson
    ```

1. Note the ID.
``````

> ## Follow up
> 
> To see how to use created Type Instances check out our [Mattermost example](mattermost-installation.md).

Reviewer: ```suggestion
## Next steps

To see how to use created TypeInstances, check out our [Mattermost example](mattermost-installation.md).
```
